src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -5985,13 +5985,13 @@ void MacroAssembler::generate_fill(BasicType t, bool aligned,
     jccb(Assembler::zero, L_skip_align2);
     movw(Address(to, 0), value);
     addptr(to, 2);
-    subl(count, 1<<(shift-1));
+    subptr(count, 1<<(shift-1));
```

Comment:
At line 5968 also we need the change from cmpl to cmpptr.
cmpl(count, 2<<shift); // Short arrays (< 8 bytes) fill by element

Reply:
Will do.